ff2mpv.py

```diff
@@ -30,13 +34,58 @@ def main():
         path = os.environ.get("PATH")
         os.environ["PATH"] = f"/usr/local/bin:{path}"
 
+    if "cookies" in message and is_whitelisted(url, get_whitelist()):
+        cookies_fname = create_cookiefile(message.get("cookies"));
+        ytdloptions["cookies"] = cookies_fname
+        additional_mpv_args += ['--cookies', '--cookies-file={}'.format(cookies_fname)]
+
+    mpv_ytdloptions = '--ytdl-raw-options-append={}'.format(
+        ",".join("{}={}".format(k,v) for k,v in ytdloptions.items()))
+
+    args = ['mpv', '--no-terminal', mpv_ytdloptions] + additional_mpv_args + ["--", url]
+
     subprocess.Popen(args, **kwargs)
 
     # Need to respond something to avoid "Error: An unexpected error occurred"
     # in Browser Console.
     send_message("ok")
 
 
+def create_cookiefile(cookies):
+    """
+    create a temporary file in netscape cookie format and return its path
+    """
+    cookiefile = tempfile.NamedTemporaryFile(mode="w+", delete=False)
+    cookiefile.write("""# Netscape HTTP Cookie File
+# http://curl.haxx.se/rfc/cookie_spec.html
+# This is a generated file!  Do not edit.
+
+""")
+    for cookie in cookies:
+        if "expirationDate" in cookie:
+            expdate = str(cookie.get("expirationDate"))
+        else:
+            expdate = "0"
+        tokens = [cookie.get("domain"),
+                  str(cookie.get("domain").startswith(".")).upper(),
+                  cookie.get("path"), str(cookie.get("secure")).upper(),
+                  expdate, cookie.get("name"),
+                  cookie.get("value")]
+        line = "\t".join(tokens)
+        cookiefile.write(line)
+        cookiefile.write("\n")
+    cookiefile.close()
+    return cookiefile.name
+
+
+def get_config_path(file=''):
```

Comment:
Let's make this a little less flexible: there's no need for a `file` keyword argument, and you can simplify the logic below:

1. If `XDG_CONFIG_HOME` is present, then expand to `$XDG_CONFIG_HOME/ff2mpv`
2. Otherwise, expand to `$HOME/ff2mpv`

For the second case, `pathlib.Path.home()` is slightly nicer than `os.path.expanduser(...)`.

Reply:
I think a user space program - especially a browser extension should **never** store its files in a non-hidden file in $HOME. And I think `$HOME/.config/` is a sensible fallback, as many programs use it as such. If you want I can additionally add `$HOME/.ff2mpv/` as fallback if `$HOME/.config/` does not exist. I know a simpler solution is easier to maintain, but in this case it comes with a significant drawback for the user.

Reply:
Whoops. Yes, you're absolutely right.

Okay, let's go with this:

1. If `XDG_CONFIG_HOME` is present: `$XDG_CONFIG_HOME/ff2mpv/`
2. Otherwise: `$HOME/.ff2mpv/`

Reply:
Ok I now implemented it like this:
1. `$XDG_CONFIG_HOME/ff2mpv/`
2. `$HOME/.config/ff2mpv/` is `$XDG_CONFIG_HOME` is not set
3. `$HOME/.ff2mpv` if `~/.config/` doesn't exist

The reason for adding `$HOME/.config/ff2mpv/` as first fallback is that I have ssen some Linux machines not having set their XDG variables right. This might especially occur if a distro doesn't handle the creation of new users properly and doesn't assign their XDG variables. However I think this is no reason to clutter up the home dir of such users, so I think checking for `~/.config/` is reasonable.

Reply:
Yes, I think the XDG_CONFIG_HOME/xyz -> ~/.confg/xyz -> ~/.xyz is the typical fallback most programs use